app/models/matchup.rb

```diff
@@ -2,33 +2,76 @@
 #
 # Table name: matchups
 #
-#  id          :bigint           not null, primary key
-#  winner_id   :bigint
-#  position    :integer
-#  bracket_id  :bigint           not null
-#  option_a_id :bigint
-#  option_b_id :bigint
-#  created_at  :datetime         not null
-#  updated_at  :datetime         not null
+#  id             :bigint           not null, primary key
+#  winner_id      :bigint
+#  position       :integer
+#  bracket_id     :bigint           not null
+#  option_a_id    :bigint
+#  option_b_id    :bigint
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#  round          :integer          default(1)
+#  round_position :integer          default(1)
 #
 class Matchup < ApplicationRecord
   belongs_to :bracket
   belongs_to :winner, class_name: "Entry", optional: true
   belongs_to :option_a, class_name: "Entry", optional: true
   belongs_to :option_b, class_name: "Entry", optional: true
 
+  after_update :send_winner_to_next_matchup, if: proc {|matchup| matchup.winner.present? }
+
   scope :untouched, -> { where(option_a.empty? && option_b.empty?) }
   scope :unplayed, -> { where(winner.empty?) }
 
-  def round
-    matchups_in_round = self.bracket.number_of_entries / 2
+  def assign_round_to_matchup
+    matchups_in_round = bracket.number_of_entries / 2
+    round = 1
+    round_position = position
+    until round_position <= matchups_in_round
+      round_position -= matchups_in_round
+      matchups_in_round /= 2
+      round += 1
+    end
+    update!(round: round)
+  end
+
+  def assign_round_position
+    matchups_in_round = bracket.number_of_entries / 2
     round = 1
-    round_position = self.position
+    round_position = position
     until round_position <= matchups_in_round
       round_position -= matchups_in_round
       matchups_in_round /= 2
       round += 1
     end
-    return round
+    update!(round_position: round_position)
+  end
+
+  def next_matchup
+    final_round = Math.log2(bracket.number_of_entries).to_i
+    next_round = round + 1
+    if round != final_round
```

Comment:
If you need to check if a particular round is the final round often enough, I might consider making a method that you can call like `.final?`